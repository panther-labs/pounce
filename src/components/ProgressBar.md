A ProgressBar should be used to indicate either progress in terms of steps that the user
needs to complete or to denote that something is not properly ready yet

A ProgressBar can have different colors:

```jsx harmony
import Box from 'components/Box';
import Flex from 'components/Flex';
import ProgressBar from 'components/ProgressBar';

<Box width={1}>
  <Box mb={6}>
    <ProgressBar progressColor="primary300" progress={0.6} />
  </Box>
  <Box>
    <ProgressBar progressColor="green200" progress={0.9} />
  </Box>
</Box>;
```

A ProgressBar can have different thickness:

```jsx harmony
import ProgressBar from 'components/ProgressBar';

<ProgressBar progressColor="primary300" thickness={2} progress={0.6} />;
```
