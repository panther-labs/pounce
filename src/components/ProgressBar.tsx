import React from 'react';
import Box from './Box';
import { Theme } from 'themes/default';

export interface ProgressBarProps {
  /** The thickness (in pixels) of the progress bar (a.k.a. vertical height). Defaults to `5` */
  thickness?: number;

  /** The color of the progress bar as it fills out */
  progressColor?: keyof Theme['colors'];

  /** A value between [0,1] denoting the progress of the bar */
  progress: number;
}

const ProgressBar: React.FC<ProgressBarProps> = ({ progress, thickness = 5, progressColor }) => {
  return (
    <Box width={1} position="relative">
      <Box height={thickness} bg="grey50" borderRadius="large" zIndex={0} />
      <Box
        position="absolute"
        borderRadius="large"
        bottom={0}
        height={thickness}
        width={progress}
        bg={progressColor}
        zIndex={1}
      />
    </Box>
  );
};

export default ProgressBar;
