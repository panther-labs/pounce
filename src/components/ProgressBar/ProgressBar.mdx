A ProgressBar should be used to indicate either progress in terms of steps that the user
needs to complete or to denote that something is not properly ready yet

A ProgressBar can have different colors:

```typescript jsx
<Flex direction="column" spacing={6} width={1}>
  <ProgressBar color="blue-400" progress={0.4} />
  <ProgressBar color="teal-400" progress={0.6} />
</Flex>
```

A ProgressBar can have different thickness:

```typescript jsx
<Flex direction="column" spacing={6} width={1}>
  <ProgressBar color="blue-400" thickness={4} progress={0.4} />
  <ProgressBar color="teal-400" thickness={6} progress={0.6} />
  <ProgressBar color="red-400" thickness={8} progress={0.8} />
</Flex>
```
