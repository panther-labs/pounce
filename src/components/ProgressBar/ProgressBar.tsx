import React from 'react';
import Box from '../Box';
import { Theme } from '../../theme';

export interface ProgressBarProps {
  /** The thickness (in pixels) of the progress bar (a.k.a. vertical height). Defaults to `5` */
  thickness?: number;

  /** The color of the progress bar as it fills out */
  color?: keyof Theme['colors'];

  /** A value between [0,1] denoting the progress of the bar */
  progress: number;
}

const ProgressBar: React.FC<ProgressBarProps> = ({ progress, thickness = 5, color }) => {
  return (
    <Box width={1} position="relative">
      <Box
        height={thickness}
        backgroundColor="white"
        backgroundOpacity={0.1}
        borderRadius="pill"
        zIndex={0}
      />
      <Box
        position="absolute"
        borderRadius="pill"
        bottom={0}
        height={thickness}
        width={progress}
        backgroundColor={color}
        transition="width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms, background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms"
        zIndex={1}
      />
    </Box>
  );
};

export default ProgressBar;
