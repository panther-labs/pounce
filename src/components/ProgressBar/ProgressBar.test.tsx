import React from 'react';
import { render } from 'test-utils';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  it('renders', async () => {
    const { container } = await render(
      <React.Fragment>
        <ProgressBar progress={0.5} />
        <ProgressBar progress={0.5} thickness={10} />
        <ProgressBar progress={0.5} color="teal-400" />
      </React.Fragment>
    );
    expect(container).toMatchSnapshot();
  });
});
