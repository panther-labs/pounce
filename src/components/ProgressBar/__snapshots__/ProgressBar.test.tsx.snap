// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders 1`] = `
.pounce-2 {
  width: 100%;
  position: relative;
}

.pounce-0 {
  height: 5px;
  background-color: white;
  border-radius: pill;
  z-index: 0;
  background-color: rgba(255,255,255,0.1);
}

.pounce-1 {
  position: absolute;
  border-radius: pill;
  bottom: 0;
  height: 5px;
  width: 50%;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
}

.pounce-3 {
  height: 10px;
  background-color: white;
  border-radius: pill;
  z-index: 0;
  background-color: rgba(255,255,255,0.1);
}

.pounce-4 {
  position: absolute;
  border-radius: pill;
  bottom: 0;
  height: 10px;
  width: 50%;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
}

.pounce-7 {
  position: absolute;
  border-radius: pill;
  bottom: 0;
  height: 5px;
  width: 50%;
  background-color: teal-400;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
}

<div>
  <div
    class="pounce-2"
  >
    <div
      class="pounce-0"
    />
    <div
      class="pounce-1"
    />
  </div>
  <div
    class="pounce-2"
  >
    <div
      class="pounce-3"
    />
    <div
      class="pounce-4"
    />
  </div>
  <div
    class="pounce-2"
  >
    <div
      class="pounce-0"
    />
    <div
      class="pounce-7"
    />
  </div>
</div>
`;
